lib/cisco_node_utils/portchannel_global.rb

```diff
@@ -0,0 +1,303 @@
+                         'symmetry')
+    end
+
+    def port_channel_load_balance=(bselect, bhash, hpoly, asy, sy, conc, rot)
+      lb_type = config_get('portchannel_global', 'load_balance_type')
+      case lb_type.to_sym
+      when :ethernet # n6k
+        if bselect == 'src'
+          sel = 'source'
+        elsif bselect == 'dst'
+          sel = 'destination'
+        else
+          sel = 'source-dest'
+        end
+        sel_hash = sel + '-' + bhash
+        # port-channel load-balance ethernet destination-mac CRC10c
+        config_set('portchannel_global', 'port_channel_load_balance',
+                   'ethernet', sel_hash, hpoly, '', '', '')
+      when :asymmetric # n7k
+        if asy == true
+          asym = 'asymmetric'
+        else
+          asym = ''
+        end
+        # port-channel load-balance dst ip-l4port rotate 4 asymmetric
+        config_set('portchannel_global', 'port_channel_load_balance',
+                   bselect, bhash, 'rotate', rot.to_s, asym, '')
+      when :symmetry # n9k
+        sym = sy ? 'symmetric' : ''
+        concat = conc ? 'concatenation' : ''
+        rot_str = rot.zero? ? '' : 'rotate'
+        rot_val = rot.zero? ? '' : rot.to_s
+        # port-channel load-balance src-dst ip rotate 4 concatenation symmetric
+        config_set('portchannel_global', 'port_channel_load_balance',
+                   bselect, bhash, rot_str, rot_val, concat, sym)
+      end
+    rescue Cisco::CliError => e
+      raise "[#{@name}] '#{e.command}' : #{e.clierror}"
+    end
+
+    # on n6k, the bundle hash and bundle select are
+    # merged into one and so we need to break them apart,
+    # also they are called source and destination instead of
+    # src and dst as in other devices, so we convert them
+    def _parse_ethernet_params(list, params)
+      hash_poly = params[2]
+      # hash_poly is not shown on the running config
+      # if it is default under some circumstatnces
+      hash_poly = hash_poly.nil? ? 'CRC10b' : hash_poly
+      select_hash = params[1]
+      lparams = select_hash.split('-')
+      if lparams[0].downcase == 'destination'
+        bselect = 'dst'
+        bhash = lparams[1]
+      else
+        if select_hash.include? '-dest-'
+          bselect = 'src-dst'
+          bhash = lparams[2]
+          # there are bundles hashes like ip-only and
+          # port-only specific to src-dst
+          bhash += '-only' if select_hash.include? 'only'
+        else
+          bselect = 'src'
+          bhash = lparams[1]
+        end
+      end
+      list << bselect << bhash << hash_poly << nil << nil << nil << nil
+    end
+
+    def _parse_asymmetric_params(list, params, line)
+      bselect = params[0]
+      bhash = params[1]
+      # asymmetric keyword does not show up if it is false
+      asym = (line.include? 'asymmetric') ? true : false
+      ri = params.index('rotate')
+      rotate = params[ri + 1].to_i
+      list << bselect << bhash << nil << asym << nil << rotate << nil
```

Comment:
While this does work, it's a good example of why I dislike the positional list for this.